gno.land/pkg/gnoland/events.go

```diff
@@ -0,0 +1,57 @@
+package gnoland
+
+import (
+	"github.com/gnolang/gno/tm2/pkg/events"
+	"github.com/rs/xid"
+)
+
+// eventType encompasses all event types
+// that can appear in the collector
+type eventType interface {
+	validatorUpdate
+}
+
+// filterFn is the filter method for incoming events
+type filterFn[T eventType] func(events.Event) []T
+
+// collector is the generic in-memory event collector
+type collector[T eventType] struct {
+	events []T         // temporary event storage
+	filter filterFn[T] // method used for filtering events
+}
+
+// newCollector creates a new event collector
+func newCollector[T eventType](
+	evsw events.EventSwitch,
+	filter filterFn[T],
+) *collector[T] {
+	c := &collector[T]{
+		events: make([]T, 0),
+		filter: filter,
+	}
+
+	// Register the listener
+	evsw.AddListener(xid.New().String(), func(e events.Event) {
```

Comment:
Is this ID sometimes printed? It would likely be better to use a human-readable name for debugging.

Reply:
It's never printed, it's used internally by the event switch to track listeners (and remove them)